core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb.go

```diff
@@ -685,8 +674,22 @@ func (vdb *VersionedDB) postCommitProcessing(committers []*committer, namespaces
 
 	}()
 
+	for _, ns := range namespaces {
+		db, err := vdb.getNamespaceDBHandle(ns)
+		if err != nil {
+			return err
+		}
+		if db.couchInstance.conf.WarmIndexesAfterNBlocks > 0 {
+			if db.indexWarmCounter >= db.couchInstance.conf.WarmIndexesAfterNBlocks {
+				go db.runWarmIndexAllIndexes()
+				db.indexWarmCounter = 0
+			}
+			db.indexWarmCounter++
+		}
+	}
+
 	// Record a savepoint at a given height
-	if err := vdb.ensureFullCommitAndRecordSavepoint(height, namespaces); err != nil {
+	if err := vdb.recordSavepoint(height, namespaces); err != nil {
```

Comment:
Out of scope for this PR. Given that `recordSavepoint` is the time-consuming operation in this function, we can run it first (i mean in the beginning of this function) in a goroutine in parallel with the cache update and warmIndex.

Reply:
Yes, I would keep any further optimization out of scope of this. BTW, warmIndex stuff already happens in a separate goroutine.